pandas/core/generic.py

```diff
@@ -683,11 +685,52 @@ def transpose(self, *args, **kwargs):
 
         new_axes = self._construct_axes_dict_from(self, [self._get_axis(x)
                                                          for x in axes_names])
-        new_values = self.values.transpose(axes_numbers)
-        if kwargs.pop('copy', None) or (len(args) and args[-1]):
-            new_values = new_values.copy()
 
+        copy = kwargs.pop('copy', None) or (len(args) and args[-1])
         nv.validate_transpose_for_generic(self, kwargs)
+
+        values = self.values
+        if (isinstance(values, ABCDatetimeIndex) and
+                values.tz is not None and self.ndim > 1):
+            # transpose is a no-op, and passing axes would raise ValueError
+            #  as the DatetimeIndex.transpose method does not accept that kwarg
+            tz = values.tz
+            utc_values = values.asi8.reshape(self.shape)
+            utc_values = utc_values.transpose(axes_numbers)
+            if copy:
+                utc_values = utc_values.copy()
+            result = self._constructor(utc_values, **new_axes)
+            if self.ndim > 2:
+                # We're assuming DataFrame from here on
```

Comment:
not sure why you should do this here at all. Move this to internals and just create new blocks, rather than hacking it.

Reply:
The point is that it is going to be a hack regardless unless we allow 2D EAs.

Reply:
not going to happen any time soon if ever

as i said this is way too hacky and needs to be pushed to internals